std/assembly/string.ts

```diff
@@ -484,6 +484,1029 @@ import { idof } from "./builtins";
+        store<u16>(out + outpos + 4, 0x0399);
+        outpos += 6;
+        continue;
+      }
+
+      // even and odd calculation for normal cases
+      let odd = bool(char & 1);
+      let even = !odd;
+
+      if (char >= 43888 && char <= 43967) char -= 38864;
+      else if (char == 11365) char = 570;
+      else if (char == 11366) char = 574;
+      else if (char == 7834) char = 65;
+      else if (char == 7830) char = 72;
+      else if (char == 7831) char = 84;
+      else if (char == 7832) char = 87;
+      else if (char == 7833) char = 89;
+      else if (char == 8180) char = 911;
+
+      else if (char >= 11520 && char <= 11557
+        || char == 11559
+        || char == 11565) char -= 7264;
+
+      else if (char == 8188
+        || char == 8183
+        || char == 8182
+        || char == 8179) char = 937;
+
+      else if (char == 8164) char = 929;
+
+      else if (char == 8162
+        || char == 8163
+        || char == 8166
+        || char == 8167) char = 933;
+
+      else if (char == 8132) char = 905;
+
+      else if (char == 8150
+        || char == 8151
+        || char == 8147
+        || char == 8146) char = 921;
+
+      else if (char == 8140
+        || char == 8135
+        || char == 8134
+        || char == 8131) char = 919;
+
+      else if (char == 8116) char = 902;
+      else if (char == 8126) char = 921;
+
+      else if (char == 8124
+        || char == 8119
+        || char == 8118
+        || char == 8115) char = 913;
+
+      else if (char == 8022
+        || char == 8020
+        || char == 8018
+        || char == 8016) char = 933;
+
+      else if (char == 7296) char = 1042;
+      else if (char == 7297) char = 1044;
+      else if (char == 7298) char = 1054;
+      else if (char == 7301) char = 1058;
+      else if (char == 7299) char = 1057;
+      else if (char == 7300) char = 1058;
+      else if (char == 7302) char = 1066;
+      else if (char == 7303) char = 1122;
+      else if (char == 43859) char = 42931;
+      else if (char == 496) char = 74;
+      else if (char == 305) char = 73;
+      else if (char == 658) char = 439;
+      else if (char == 640) char = 422;
+      else if (char == 643) char = 425;
+      else if (char == 648) char = 430;
+      else if (char == 650) char = 433;
+      else if (char == 651) char = 434;
+      else if (char == 629) char = 415;
+      else if (char == 626) char = 413;
+      else if (char == 617) char = 406;
+      else if (char == 623) char = 412;
+      else if (char == 595) char = 385;
+      else if (char == 616) char = 407;
+      else if (char == 611) char = 404;
+      else if (char == 596) char = 390;
+      else if (char == 598) char = 393;
+      else if (char == 599) char = 394;
+      else if (char == 608) char = 403;
+      else if (char == 603) char = 400;
+      else if (char == 601) char = 399;
+
+      else if (char == 383
+        || char == 223
+        || char == 64261
+        || char == 64262) char = 83;
```

Comment:
For `64261` and `64262` this not correct. They are return surrogate pair which only one code point contain `83`. I guess we have a lot of other similar issues. Plz recheck